Src/WindowsServer/WindowsServer.Shared.Tests/AppServicesHeartbeatTests.cs

```diff
@@ -0,0 +1,124 @@
+            foreach (var envVarKvp in envVars)
+            {
+                string newVal = string.Concat(envVarKvp.Value, "_1");
+                Environment.SetEnvironmentVariable(envVarKvp.Key, newVal);
+            }
+
+            envVars = GetEnvVarsAssociatedToModule();
+
+            Assert.True(appSrvHbeatModule.UpdateHeartbeatWithAppServiceEnvVarValues());
+            foreach (var kvp in AppServicesHeartbeatTelemetryModule.WebHeartbeatPropertyNameEnvVarMap)
+            {
+                Assert.True(hbeatProviderMock.HbeatProps.ContainsKey(kvp.Key));
+                Assert.Equal(hbeatProviderMock.HbeatProps[kvp.Key], envVars[kvp.Value]);
+            }
+        }
+
+        [TestMethod]
+        public void NoHeartbeatManagerAvailableDoesntThrow()
+        {
+            var envVars = GetEnvVarsAssociatedToModule();
+            var appSrvHbeatModule = new AppServicesHeartbeatTelemetryModule();
+            try
+            {
+                appSrvHbeatModule.Initialize(null);
+            }
+            catch (Exception any)
+            {
+                Assert.False(any == null);
+            }
+        }
+
+        [TestMethod]
+        public void NoAppServicesEnvVarsWorksWithoutFailure()
+        {
+            var envVars = GetEnvVarsAssociatedToModule();
+            // ensure all environment variables are set to nothing (remove them from the environment)
+            foreach (var kvp in envVars)
+            {
+                Environment.SetEnvironmentVariable(kvp.Key, string.Empty);
+            }
+            var hbeatProviderMock = new HeartbeatProviderMock();
+            var appSrvHbeatModule = new AppServicesHeartbeatTelemetryModule(hbeatProviderMock);
+            Assert.True(appSrvHbeatModule.UpdateHeartbeatWithAppServiceEnvVarValues());
+            foreach (var kvp in AppServicesHeartbeatTelemetryModule.WebHeartbeatPropertyNameEnvVarMap)
+            {
+                Assert.False(hbeatProviderMock.HbeatProps.ContainsKey(kvp.Key));
+            }
+        }
+
+        /// <summary>
+        /// Return a dictionary containing the expected environment variables for the AppServicesHeartbeat module. If
+        /// the environment does not contain a value for them, set the environment to have them.
+        /// </summary>
+        /// <returns>Dictionary with expected environment variable names as the key, current environment variable content as the value.</returns>
+        private Dictionary<string,string> GetEnvVarsAssociatedToModule()
+        {
+            Dictionary<string, string> envVars = new Dictionary<string, string>();
+            foreach (var kvp in AppServicesHeartbeatTelemetryModule.WebHeartbeatPropertyNameEnvVarMap)
+            {
+                envVars.Add(kvp.Value, Environment.GetEnvironmentVariable(kvp.Value));
+                if (string.IsNullOrEmpty(envVars[kvp.Value]))
+                {
+                    Environment.SetEnvironmentVariable(kvp.Value, kvp.Key);
```

Comment:
The effect is setting the ENV variable persists even after the test? This may have unintended consequences in the next test. 
If modifying any ENV, make sure to set it back to original after the test

Reply:
Agreed. Also, in PR #870 see that I've added a ClassInitializer/ClassCleanup method to handle environment changes.

Reply:
Merged #870 here where I fixed this issue by using unique env vars in place of the standard during testing. Note that I left the original testing for the TelemetryInitializer intact (although it really should be updated to do the same).